components/custom_status/custom_status_emoji.tsx

```diff
@@ -4,31 +4,40 @@ import React from 'react';
 import {Tooltip} from 'react-bootstrap';
 import {useSelector} from 'react-redux';
 
-import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
-
 import OverlayTrigger from 'components/overlay_trigger';
 import RenderEmoji from 'components/emoji/render_emoji';
-
-import Constants from 'utils/constants';
+import {getCustomStatus, isCustomStatusEnabled} from 'selectors/views/custom_status';
 import {GlobalState} from 'types/store';
+import Constants from 'utils/constants';
 
 interface ComponentProps {
     emojiSize?: number;
     showTooltip?: boolean;
     tooltipDirection?: 'top' | 'right' | 'bottom' | 'left';
+    emojiStyle?: React.CSSProperties;
+    userID?: string;
 }
 
-const CustomStatusEmoji = ({tooltipDirection = 'bottom', showTooltip = false, emojiSize = 16}: ComponentProps) => {
-    const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
-    if (!(currentUser && currentUser.props && currentUser.props.customStatus)) {
+const CustomStatusEmoji = (props: ComponentProps) => {
+    const {emojiSize, emojiStyle, showTooltip, tooltipDirection, userID} = props;
+    const customStatusEnabled = useSelector((state: GlobalState) => {
+        return isCustomStatusEnabled(state);
+    });
```

Comment:
If `isCustomStatusEnabled` is an arrow function, would this work?

```suggestion
    const customStatusEnabled = useSelector(isCustomStatusEnabled);
```

If so, make similar changes elsewhere.

Reply:
`isCustomStatusEnabled` is not an arrow function.